server/sonar-server/src/test/java/org/sonar/server/rule/ws/SearchActionTest.java

```diff
@@ -177,31 +295,400 @@ public void should_return_specified_fields() throws IOException {
+      p.setDefaultValue("some value")
+        .setType("STRING")
+        .setDescription("My small description")
+        .setName("my_var"));
+
+    RuleActivation activation = RuleActivation.create(rule.getKey());
+    List<ActiveRuleChange> activeRuleChanges = ruleActivator.activate(dbTester.getSession(), activation, profile);
+
+    // Insert directly in database a rule parameter with a null value
+    ActiveRuleParamDto activeRuleParam = ActiveRuleParamDto.createFor(ruleParam).setValue(null);
+    dbTester.getDbClient().activeRuleDao().insertParam(dbTester.getSession(), activeRuleChanges.get(0).getActiveRule(), activeRuleParam);
+
+    dbTester.commit();
+
+    indexRules();
+    indexActiveRules();
+
+    SearchResponse result = ws.newRequest()
+      .setParam("f", "actives")
+      .setParam("q", rule.getName())
+      .setParam("organization", organization.getKey())
+      .setParam("activation", "true")
+      .setParam("qprofile", profile.getKee())
+      .executeProtobuf(SearchResponse.class);
+
+    assertThat(result.getTotal()).isEqualTo(1);
+    assertThat(result.getRulesCount()).isEqualTo(1);
+    assertThat(result.getActives()).isNotNull();
+    assertThat(result.getActives().getActives().get(rule.getKey().toString())).isNotNull();
+    assertThat(result.getActives().getActives().get(rule.getKey().toString()).getActiveListList()).hasSize(1);
+
+    Rules.Active activeList = result.getActives().getActives().get(rule.getKey().toString()).getActiveList(0);
+    assertThat(activeList.getParamsCount()).isEqualTo(2);
+    assertThat(activeList.getParamsList()).extracting("key", "value").containsExactlyInAnyOrder(
+      tuple(ruleParam.getName(), ruleParam.getDefaultValue()),
+      tuple(activeRuleParam.getKey(), "")
+    );
+  }
+
+  @Test
+  public void statuses_facet_should_be_sticky() throws Exception {
+    RuleDefinitionDto rule1 = dbTester.rules().insert(r -> r.setLanguage("java"));
+    RuleDefinitionDto rule2 = dbTester.rules().insert(r -> r.setLanguage("java").setStatus(RuleStatus.BETA));
+    RuleDefinitionDto rule3 = dbTester.rules().insert(r -> r.setLanguage("java").setStatus(RuleStatus.DEPRECATED));
+
+    indexRules();
+
+    SearchResponse result = ws.newRequest()
+      .setParam("f", "status")
+      .setParam("status", "DEPRECATED")
+      .executeProtobuf(SearchResponse.class);
+
+    assertThat(result.getRulesCount()).isEqualTo(3);
+    assertThat(result.getRulesList()).extracting("key", "status.name").containsExactlyInAnyOrder(
+      tuple(rule1.getKey().toString(), rule1.getStatus().name()),
+      tuple(rule2.getKey().toString(), rule2.getStatus().name()),
+      tuple(rule3.getKey().toString(), rule3.getStatus().name())
+    );
+  }
+
+  @Test
+  public void statuses_facet_should_have_all_statuses_except_removed() throws Exception {
+    SearchResponse result = ws.newRequest()
+      .setParam("f", "status")
+      .executeProtobuf(SearchResponse.class);
+    assertThat(result.getFacets());
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Complete the assertion. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2970)